client/lcd/certificates.go

```diff
@@ -0,0 +1,163 @@
+package lcd
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"os"
+	"strings"
+	"time"
+)
+
+// default: 10 years
+const defaultValidFor = 365 * 24 * time.Hour
+
+func generateSelfSignedCert(host, ecdsaCurve string, rsaBits int) (certBytes []byte, priv interface{}, err error) {
+	switch ecdsaCurve {
```

Comment:
I'm not sure that this should be configureable. P-256 keys should be fine for our use case

Reply:
ACKd, amended.